src/net_processing.cpp

```diff
@@ -5155,6 +5157,12 @@ void PeerManagerImpl::EvictExtraOutboundPeers(std::chrono::seconds now)
             if (state == nullptr) return; // shouldn't be possible, but just in case
             // Don't evict our protected peers
             if (state->m_chain_sync.m_protect) return;
+            // Protect a peer if it's the only one for its network, counting IPv4 and
+            // IPv6 as one network.
+            // Both outbound-full-relay and manual connections are counted for this,
+            // but not block-relay-only connections because the goal
+            // is to have at least one tx-relaying connection to each reachable network.
+            if (m_connman.GetFullOutboundAndManualCount(pnode->addr.GetNetwork()) <= 1) return;
```

Comment:
We want to protect a peer if it's the only one for its network. However, we can run a node supporting just one network. In this case, I suppose that calling `GetFullOutboundAndManualCount` might be unnecessary, we would iterating the whole `m_nodes` with no need.

In this case, I think we could have a function like `SupportsMultipleNetworks` to check whether we support more than one network, e.g.:
```cpp
bool CConnman::SupportsMultipleNetworks() const
{
    int count = 0;
    for (int n = 0; n < NET_MAX; n++) {
        enum Network net = (enum Network)n;
        if (net == NET_UNROUTABLE || net == NET_INTERNAL) continue;
        if (IsReachable(net)) {
            if (net == NET_IPV4) n++;
            count++;
        }
    }
    return count > 0;
}
```

if so, we can call `GetFullOutboundAndManualCount`:
```cpp
if (m_connman.SupportsMultipleNetworks() && m_connman.GetFullOutboundAndManualCount(pnode->addr.GetNetwork()) <= 1) return;
```